packages/neon_framework/lib/src/bloc/bloc.dart

```diff
@@ -35,7 +35,7 @@ abstract class InteractiveBloc extends Bloc {
   /// Refreshes the state of the bloc.
   ///
   /// Commonly involves re-fetching data from the server.
-  FutureOr<void> refresh();
+  Future<void> refresh();
```

Comment:
The problem is that FutureOr doesn't play nice with tear-offs and other async code. I'll push a second commit but you will see that it is very ugly. I'd just go with Future to avoid this

Reply:
I fear that this will not play nice when interpreted (like in an js environment). Maybe the compiler can optimize this :thinking: 
I think that we have two solutions.
1. add a separate synchronous method like `backgroundRefresh()`
2. just let it return a future and ignore the lint error. When thinking about the bloc pattern we have a stream of events that lead to a different stream of events being fired (just a recap). Adding an event to a stream happen synchronously so there should be no reason to actually await it. The emitted states should already handle changes in the UI. If somebody really needs to await the event this means that business logic is located in the UI code which is what we try to avoid in the first place.

Reply:
The problem is we rely on the future for the refresh indicator in lists. There is no easy way to make the refresh indicators stateful so that we can hook them to the streams. We could also just ignore that and make the function a void and then the refresh indicators always hide instantly because there's nothing to await. I prefer that as we already have the linear progress indicator to show that something is loading at the moment.

Reply:
ok let's keep it until we tackle https://github.com/nextcloud/neon/issues/800